### PR TITLE
feat: present the html provided insights

### PR DESCRIPTION
**Pull request by egdbear**
The type of this PR is: **Feat**

This PR solves [DIA-179]

### Description
It uses the HTML generated insights.
[Metaphysics PR](https://github.com/artsy/metaphysics/pull/5312)

Screenshot 2023-09-29 at 09 49 37

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DIA-179]: https://artsyproduct.atlassian.net/browse/DIA-179?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dzucconi** [comment]
Ah I see your plan now. But I still don't love it for two reasons though: getting HTML when you aren't expecting is a problem, and we don't want to get in the habit of timing sync deploys. I'd still recommend the format arg.

File: src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx [hunk 1]
```diff
@@ -285,4 +291,11 @@ const CV = styled(RouterLink)`
   }
 `
 
+// remove paragraph margins from markdown
```
Comment (egdbear) [inline src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:294]: This is "marked" doing and I'm afraid hardcoding this setting in MP that will produce breaking changes. Open for suggestions @dzucconi @jonallured

Comment (dzucconi) [reply src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:294]: We wouldn't want to hardcode this setting. This is normal and fine. Theoretically other tags with default margins could show up here but this is fine for now.

File: src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx [hunk 2]
```diff
@@ -10,11 +11,15 @@ interface ArtistCareerHighlightProps {
...
-          ? insight.entities.join(", ").replace(/,\s([^,]+)$/, ", and $1")
-          : insight.description}
-      </Text>
+      <Description
+        variant="xs"
+        dangerouslySetInnerHTML={{
+          __html:
+            insight.entities.length > 0
+              ? insight.entities.join(", ")
+              : insight.description ?? "",
```
Comment (dzucconi) [inline src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx:20]: Ah, can we use the `ArtistCareerHighlight` component here as well?

File: src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx [hunk 3]
```diff
@@ -203,11 +203,17 @@ const ArtistHeader: React.FC<ArtistHeaderProps> = ({ artist }) => {
...
-                    </Text>
+                    <Description
+                      variant="sm"
+                      color="black60"
+                      pb={1}
+                      dangerouslySetInnerHTML={{
+                        __html:
+                          insight.entities.length > 0
+                            ? insight.entities.join(", ")
+                            : insight.description ?? "",
```
Comment (dzucconi) [inline src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:214]: Can we use the `ArtistCareerHighlight` component here as well?

File: src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx [hunk 4]
```diff
@@ -10,11 +11,15 @@ interface ArtistCareerHighlightProps {
...
+        }}
+      ></Description>
```
Comment (dzucconi) [inline src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx:22]: Self-closing

File: src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx [hunk 5]
```diff
@@ -10,11 +11,15 @@ interface ArtistCareerHighlightProps {
 const ArtistCareerHighlight: FC<ArtistCareerHighlightProps> = ({ insight }) => {
   return (
     <Expandable label={insight.label} pb={1}>
-      <Text variant="xs">
-        {insight.entities.length > 0
-          ? insight.entities.join(", ").replace(/,\s([^,]+)$/, ", and $1")
```
Comment (dzucconi) [inline src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx:15]: We don't want to get rid of this to sentence bit.

File: src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx [hunk 6]
```diff
@@ -10,11 +11,15 @@ interface ArtistCareerHighlightProps {
...
-          : insight.description}
-      </Text>
+      <Description
+        variant="xs"
+        dangerouslySetInnerHTML={{
+          __html:
+            insight.entities.length > 0
+              ? insight.entities.join(", ")
+              : insight.description ?? "",
+        }}
```
Comment (dzucconi) [inline src/Apps/Artist/Routes/Overview/Components/ArtistCareerHighlight.tsx:21]: You'll want to do something like:

```
{...(insight.description
          ? {
              dangerouslySetInnerHTML: {
                __html: insight.description,
              },
            }
          : {
              children: insight.entities
                .join(", ")
                .replace(/,\s([^,]+)$/, ", and $1"),
            })}
```

File: src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx (hunk 3 above)
Comment (dzucconi) [reply src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:214]: I can't see the upload but I'm assuming the answer is: use the `ArtistCareerHighlightFragmentContainer` and the associated fragment.

File: src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx [hunk 7]
```diff
@@ -235,10 +228,7 @@ export const ArtistHeaderFragmentContainer = createFragmentContainer(
           credit
         }
         insights {
-          kind
```
Comment (dzucconi) [inline src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:238]: You still need `kind` — surprised this didn't type error?

**Review by dzucconi — DISMISSED** [review]
I still think you need `kind` for the `key` (these things _really_ should have IDs, surprised these cache correctly). Otherwise looks good.

File: src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx (hunk 7 above)
Comment (egdbear) [reply src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:238]: Its in `ArtistCareerHighlight_insight`

Comment (dzucconi) [reply src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:238]: That should absolutely type error — you cannot reach into fragments — lemme pull down and see what's going on

Comment (egdbear) [reply src/Apps/Artist/Components/ArtistHeader/ArtistHeader2.tsx:238]: WOW

**Review by dzucconi — DISMISSED** [review]

**Pull request merged**